### ISSUE TITLE
JSSv3 - WIP

### ISSUE THREAD

**Issue by mochet** [Joystream/community-repo#774]

**Comment by bedeho** [comment]
I still think this description sort of lacks a _vision_. What is the role of this product? who is it for? what problem does it solve? If you want to get buyin from future contributors, I would try to formulate that very clearly. I am not saying this issue itself should be the pitch, you can do that in other places, but the essence of the vision that serves as a foundation for that fits in very well here.

Even I who have discussed this question about the role of JSSv3 does not recall or properly understand where it is going, so you are well served reiterating this.

> Previously we weren't concerned with the FM program at all, now that is the primary concern and we should aim to address that first and then build upon it with expanded views of on-chain data from the QN

Is this a wise goal, given that program will end August?

**Comment by mochet** [comment]
@bedeho 
> Is this a wise goal, given that program will end August?
Absolutely, we are suffering extremely from a lack of visibility. I get the timeframe with the FM program, however this is explored in consideration of that, in fact the design has been specifically chosen to solve that problem.
I had drawn out plans for JSS in JSSv2 (https://github.com/Joystream/community-repo/issues/518) long, long ago but it is pure coincedence that the design of not only JSS but also the FM program match on enough common parameters and considerations that it only makes sense to go full blast with this, as strange as it may seem.

I would heavily suggest reading the section within this issue titled `Value proposition`: https://github.com/Joystream/community-repo/issues/792

There are very, very long term things in terms of value and functionality that this is all aiming to deliver upon extremely quickly. While we _could_ wait for mainnet and delivery to make several of these things happen, the reality is that extensive knowledge of our DAO/runtime is quite limited. Instead of relying up on the few people who know this stuff well, I aim for us to give the answers ahead of time and put it out there before people even have to ask.

**Comment by mochet** [comment]
> Even I who have discussed this question about the role of JSSv3 does not recall or properly understand where it is going, so you are well served reiterating this.

I did a run through of some Block Explorer projects and looked at how they showed information from either Polkadot mainnet or parachains and I think this effectively proves the value that can easily be obtained by JSSv3, with the following key points:
A) Polkadot recently was worth $30bn USD+, its currently worth $8bn USD. Either way here are the 3rd party tools available to understand Polkadot
B) Besides running public infrastructure, when it comes to interpreting our data it seems that no one else is going to fix this quick enough for it to be valuable.
C) We have already demonstrated better views of our DAO via `Joystreamstats v1` and a Google Spreadsheet + charts than any available option (to the best of my awareness)

#### Example A: Subscan showing Nodle which is an official parachain

1. Has only one option for Governance which only shows Tech. comm. Proposals--nothing else
2. Go to extrinsics page and you get a bunch of generic transactions, none of these show value

Reference link: https://nodle.subscan.io/extrinsic

#### Example B: Subscan showing Altair which won a parachain auction on Kusama

1. Has a few options for governance
2. When it comes to Council Motions it doesn't have a human readable title or description of any motion. Therefore it has limited use.
3. Treasury proposals is the exact same

Link: https://altair.subscan.io/council

#### Example C: Polkadot mainnet on Subscan

1. Shows description for bounties
2. Can't show historic bounties
3. Shows proposer ID, I think that is linked to their on-chain identity module but doesn't show it for all
4. Doesn't show USD equiv.
5. The Status is unclear and therefore useless--no human understands or cares what extended as a status means, therefore it doesn't suggest any value in the Polkadot project

Link: https://polkadot.subscan.io/bounty

#### Example D: Polkadot mainnet on Polkascan

1. Dashboard view is a bit limited, just a cool visualization of blocks being printed: https://explorer.polkascan.io/polkadot
2. Takes a `very long time` to load
3. No views of governance/elections/identity
4. Extrinsics view can only allow you to look at extrinsics submitted--PROVIDED you actually know what extrinsics it allows. ZERO are human readable

Reference link: https://explorer.polkascan.io/

#### Example E: Polkadot mainnet on dotscanner

1. ZERO governance views
2. ZERO collation of extrinsics (although they do have nice labels--except you cannot click them)
3. No association with USD/human readable value and Latest Transfers

Reference link: https://dotscanner.com/polkadot/search


#### Example F: Polkastats showing Polkadot mainnet

1. No governance views
2. No USD valuation
3. No easy grouping of extrinsics
4. No way to discern value of extrinsics
5. Does have a staking dashboard, which is of limited value. Joystreamstats already did a better job of this.
6. We already did far better than this does, more than a year ago.

Reference link: https://polkastats.io/events
